### DISCUSSION TITLE
Error when passing multiple tags to the `short-url:create` console command

### DISCUSSION THREAD

**Discussion by mdrxy** [shlinkio/discussions#149] [Help wanted] [answered]
### Summary

I think that in the option description for short-url tags, it should be indicated that multiple values are to be comma separated. 

"Tags to apply to the short URL (multiple values allowed)" would change to "Tags to apply to the short URL (multiple comma-separated values allowed)"

### Use case

Increases accessibility, reduces confusion and likelihood of `Too many arguments to "short-url:create" command, expected arguments "longUrl".` error for people who assume space separated.

**Comment by acelaya** [comment]
Can you be more specific on where do you see that description?

**Comment by mdrxy** [comment]
@acelaya running `docker exec -it shlink-container shlink short-url:create -h`

**Comment by acelaya** [comment, marked as answer]
And how did you run the command? Supposedly `shlink short-url:create --tag foo --tag bar --tag baz` should work and add the three tags to the short URL.

Supporting comma-separated tags (as in `--tag foo,bar,baz`) is there only for backwards compatibility, but it is not the expected approach.

Does the approach above fail for you in any way?

**Comment by acelaya** [comment]
By looking at the error again, I have the suspicion you did something like this: `shlink short-url:create https://example.com --tag foo bar baz`.

That's unfortunately never going to work, because spaces have a special meaning in the CLI. In this case, the command parser is interpreting every "word" which is not preceded by a flag as an individual argument.

Since this command only expects one "free" argument (the long URL), it throws that error.

In console commands, when a flag is an array that accepts multiple values, you need to prepend the flag identifier (or the shortcut, in this case `-t`) to every value.

**Reply by mdrxy** [reply]
Here's how the command was ran:

```
docker exec -it {container name} shlink short-url:create -t {tag1} {tag2} --title "{title}" {longURL}
```

`-t {tag1} -t {tag2}` makes sense, though I guess I've used different programs where repeating the flag wasn't necessary for each value, so I didn't initially try that. 

I still think that somewhere in the docs, if it isn't already, it may be nice to explicitly specify how users are expected to define multiple tags, since it is rare (at least in our use) to use only one tag.
